exasol/analytics/schema/statements.py

```diff
@@ -0,0 +1,129 @@
+from __future__ import annotations
+
+from typing import Any
+
+from exasol.analytics.schema.column import Column
+from exasol.analytics.schema.column_name import ColumnName
+from exasol.analytics.schema.values import quote_value
+
+
+class UnknownColumnError(Exception):
+    """
+    In case of adding a value for a column not contained in the initial
+    list of available columns.
+    """
+
+
+class InsertStatement:
+    """
+    Enables adding columns and values for creating an SQL INSERT statement.
+
+    All available columns need to be specified up front when creating an
+    instance of this class.
+
+    Trying to add a value for a column missing in the initial list will raise
+    an exception.
+
+    Let's look at an example for an INSERT statement:
+
+        INSERT INTO "T" ("C1", "C2", "C3")
+        SELECT SYSTIMESTAMP(), 'Phase', SQ.R
+        FROM VALUES (1)
+        CROSS JOIN (SELECT count(1) as R FROM S.T) as SQ
+
+    Here we see 3 columns "C1", "C2", and "C3" being inserted into table "T"
+    while the values after `SELECT` have 3 different categories:
+
+    * SYSTIMESTAMP() is an SQL scalar function that must not be quoted.
+    * "Phase" is a string constant, that must be enclosed in single-quotes.
+    * SQ.R is a reference to column "R" in a subquery with the alias "SQ".
+
+    The alias "R" can be found again inside the subquery `(SELECT count(1) as
+    R FROM S.T)` as an alias for the `count(1)`. Alias "SQ" is assigned to the
+    subquery at the very end of the INSERT statement.
+
+    Use methods `add_constants()`, `add_scalar_functions()`, and
+    `add_references()` to add constants, scalar functions, and reference to
+    columns of the subquery, respectively.
```

Comment:
This entirely optional comment. Insead of the previous paragraph, that states what someone with SQL skills can observe, I would complete the example above example.

```
my_insert_columns = [
   Column(ColumnName('C1'), ColumnType(''TIMESTAMP')),
   Column(ColumnName('C2'), ColumnType('VARCHAR', size=1000)),
   Column(ColumnName('C3'), ColumnType('INTEGER))
]
my_statement = InsertStatement(my_insert_columns)
my_statement.add_scalar_functions({'C1': 'SELECT SYSTIMESTAMP()'}
my_statement.add_constants({'C2': 'Phase'})
my_statement.add_references(...)
```

Reply:
Should I add this example to the docstring?

Reply:
It's up to you, but I think it would help. It would need to be completed though.